src/com/amazon/corretto/crypto/provider/Utils.java

```diff
@@ -422,6 +426,18 @@ static void zeroByteBuffer(ByteBuffer buffer) {
         }
     }
 
+    private static KeyFactory getKeyFactory(AmazonCorrettoCryptoProvider provider, String algorithm) throws NoSuchAlgorithmException {
+        // Manual listing of supported algorithms for fast-path KeyFactory creation
+        switch (algorithm) {
+            case "RSA":
+                return provider.getKeyFactory(EvpKeyType.RSA);
+            case "EC":
+                return provider.getKeyFactory(EvpKeyType.EC);
+            default:
+                return KeyFactory.getInstance(algorithm);
```

Comment:
minor optional suggestion: bonus points if we can construct this reflectively (looping over EvpKeyType values, constructing a map, get'ing if present else return KeyFactory.getInstance

that map could be constructed statically to avoid paying for reflection on every call.

Reply:
this thread only amounts to a nit, but my concern here is more future-proofing than clarity -- when future programmers add a new `EvpKeyType`, how will they know to come update this function?

this is a static method. why can't we construct the aforementioned map statically over the values of `EvpKeyType` and then use it here?

Reply:
As long as we continue to lazily construct the factories, we can do that.

I have an idea for an easy refactor. Let me get it done.

Reply:
LGTM!